modules/Core/src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryScreen.java

```diff
@@ -43,4 +61,103 @@ public EntityRef get() {
     public boolean isModal() {
         return false;
     }
+
+    /*
+      The numbersBetween() and getTransferEntity() methods were
+      originally in the InventoryCell class. They were copied over
+      to here because they are private functions that the
+      moveItemSmartly() method needs to function. The first section
+      of code in onClosed() is based on the moveItemSmartly() method.
+    */
+    private List<Integer> numbersBetween(int start, int exclusiveEnd) {
+        List<Integer> numbers = new ArrayList<>();
+        for (int number = start; number < exclusiveEnd; number++) {
+            numbers.add(number);
+        }
+        return numbers;
+    }
+
+    private EntityRef getTransferEntity() {
+        return CoreRegistry.get(LocalPlayer.class).getCharacterEntity().getComponent(CharacterComponent.class).movingItem;
```

Comment:
Three more sneaky usages of CoreRegistry hiding in this class. Plus a couple imports that could be removed.

There may be some cases in which `@In` would initialize a local service too early meaning it then won't work, but give it a try and see if it works